src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs

```diff
@@ -265,45 +266,79 @@ public void ComputeReturnValueTreatment(CorElementType type, TypeHandle thRetTyp
-                                    break;
+                                if (descriptor.eightByteCount == 1)
+                                {
+                                    if (descriptor.eightByteClassifications0 == SystemVClassificationType.SystemVClassificationTypeSSE)
+                                    {
+                                        fpReturnSize = sizeof(double);
+                                    }
+                                }
+                                else
+                                {
+                                    fpReturnSize = 16;
+                                    if (descriptor.eightByteClassifications0 == SystemVClassificationType.SystemVClassificationTypeSSE)
+                                    {
+                                        fpReturnSize += 1;
+                                    }
+
+                                    if (descriptor.eightByteClassifications0 == SystemVClassificationType.SystemVClassificationTypeSSE)
+                                    {
+                                        fpReturnSize += 2;
+                                    }
+                                }
 
-                                default:
-                                    throw new NotImplementedException();
+                                break;
                             }
-                            break;
                         }
-
-                        uint size = (uint)thRetType.GetSize();
-
-                        if (IsX86 || IsX64)
+                        else
                         {
-                            // Return value types of size which are not powers of 2 using a RetBuffArg
-                            if ((size & (size - 1)) != 0)
+                            if (thRetType.IsHFA() && !isVarArgMethod)
                             {
-                                usesRetBuffer = true;
+                                CorElementType hfaType = thRetType.GetHFAType();
+
+                                switch (Architecture)
+                                {
+                                    case TargetArchitecture.ARM:
+                                        fpReturnSize = (hfaType == CorElementType.ELEMENT_TYPE_R4) ?
+                                            (4 * (uint)sizeof(float)) :
+                                            (4 * (uint)sizeof(double));
+                                        break;
+
+                                    case TargetArchitecture.ARM64:
+                                        // DESKTOP BEHAVIOR fpReturnSize = (hfaType == CorElementType.ELEMENT_TYPE_R4) ? (4 * (uint)sizeof(float)) : (4 * (uint)sizeof(double));
+                                        // S and D registers overlap. Since we copy D registers in the UniversalTransitionThunk, we'll
+                                        // thread floats like doubles during copying.
```

Comment:
Looks like a typo "thread" -> "treat" unless I'm mistaken.

Reply:
I have not touched this code (it just got indented), but thanks for noticing it, I'll fix it.